org.lflang/src/org/lflang/generator/ts/TSReactionGenerator.kt

```diff
@@ -103,43 +107,66 @@ class TSReactionGenerator(
         reaction: Reaction,
         reactPrologue: String,
         reactEpilogue: String,
-        reactFunctArgs: StringJoiner,
+        reactFuncArgs: StringJoiner,
         reactSignature: StringJoiner
     ): String {
         // Assemble reaction triggers
+        var reactCode = ""
+        var reactDeadline = "}"
         val reactionTriggers = StringJoiner(",\n")
-        for (trigger in reaction.triggers) {
-            if (trigger is VarRef) {
-                reactionTriggers.add(trigger.generateVarRef())
-            } else if (trigger is BuiltinTriggerRef) {
-                when (trigger.type) {
-                    BuiltinTrigger.STARTUP  -> reactionTriggers.add("this.startup")
-                    BuiltinTrigger.SHUTDOWN -> reactionTriggers.add("this.shutdown")
-                    else -> {}
+
+        if (reaction.isMutation()) {
```

Comment:
Why are these cases handled separately? They look the same to me...

Reply:
Yes, I deleted the duplicate code. https://github.com/lf-lang/lingua-franca/pull/1318/commits/0d49d6e37ef948cd22628394e533ffab613e524b